### PR TITLE
[16.0][MIG] purchase_invoice_plan: Migration to 16.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
Standard Migration
Supersede of https://github.com/OCA/purchase-workflow/pull/1880

@qrtl

### CODE REVIEW

**Review by rinaldifirdaus — APPROVED** [review]
Functional Test in Runboat: Works as expected.
the issue from v15 which occurs when adjust qty or add line on PO included in this PR. Thanks!

File: purchase_invoice_plan/models/purchase.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        if lines:
+            installments = [str(x) for x in lines.mapped("installment")]
+            raise UserError(
+                _(
+                    "Installment %s: already used and not allowed to delete.\n"
+                    "Please discard changes."
+                )
+                % ", ".join(installments)
+            )
+        return super().unlink()
```
Comment (yostashiro) [inline purchase_invoice_plan/models/purchase.py]: Let's switch to use @api.ondelete. https://github.com/OCA/pylint-odoo/pull/458

File: purchase_invoice_plan/models/purchase.py [hunk 2]
```diff
@@ -313,7 +313,8 @@ def _get_plan_qty(self, order_line, percent):
         plan_qty = order_line.product_qty * (percent / 100)
         return plan_qty
 
-    def unlink(self):
+    @api.ondelete(at_uninstall=False)
+    def unlink_installments(self):
```
Comment (yostashiro) [inline purchase_invoice_plan/models/purchase.py]: Let's make it a private method and adjust the name, following suit to other cases I can see in the standard modules.

```suggestion
    def _unlink_except_no_edit(self):
```

**Review by yostashiro — APPROVED** [review]
Code review.

**Comment by AungKoKoLin1997** [comment]
@purchase-workflow-maintainers
Can we get this PR merged?

**Comment by HviorForgeFlow** [comment]
Solving issue here: https://github.com/OCA/purchase-workflow/pull/2005

**Comment by AungKoKoLin1997** [comment]
@HviorForgeFlow Can you please try again?

**Pull request merged**